SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -401,14 +394,27 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	// If we're the client, we can now ask the server to spawn our controller.
-	if (!IsServer())
+	if (IsServer())
+	{
+		if (GlobalStateManager && !ServerConnection)
+		{
+			GlobalStateManager->SetCanBeginPlay(true);
+			GlobalStateManager->TriggerBeginPlay();
+			GlobalStateManager->SetAcceptingPlayers(true);
+		}
+	}
+	else
 	{
+		// If we're the client, we can now ask the server to spawn our controller.
 		// If we know the GSM is already accepting players, simply spawn.
-		if (GlobalStateManager->bAcceptingPlayers && GetWorld()->RemovePIEPrefix(GlobalStateManager->DeploymentMapURL) == GetWorld()->RemovePIEPrefix(GetWorld()->URL.Map))
+		UWorld* World = GetWorld();
```

Comment:
Should we check the World is valid in some way?

Reply:
You mean a nullptr check or that the data in the world is valid?

Reply:
I checked for nullptr but can add another check if needed.